### PR TITLE
Gifs to memes - problem 2 update

### PR DESCRIPTION
**Pull request by hakierka**
Clean out PR where gifs are globally changed to memes, updated problem copy for 2 and 3.

### CODE REVIEW

File: problems/4-infinite-compression/problem.md [hunk 1]
```diff
@@ -1,22 +1,42 @@
...
-
-The DAG is packed into a CAR 🚗 - beep beep. jk jk a CAR is a Content-Addressed Archive. It's kinda like a `tar` file but for DAGs. It's a useful container format for transferring content addressed data over HTTP.
-
-In web3.storage, we content address CARs as well! So we generate a hash for the CAR and invoke a `store/add` capability, passing the CAR CID as parameters to the invocation. The service will reply with a signed URL - a place to put the CAR that will only accept data that hashes to the CID of the CAR.
-
-**BUT** if we know the CAR you want to send has already been stored then we'll send back a flag to say that you don't need to send it! Hey presto, infinite compression! Joking aside, isn't content addressing awesome?!
```
Comment (alanshaw) [inline problems/4-infinite-compression/problem.md:13]: I feel like we've lost information here.

"Infinite compression" is not real, but is an attempt to put a name to this advantage:

We content address the data, and because we do that, the service can determine if it already has the data without actually sending it. If the service has the data, you do not need to send it - i.e. you have "infinite compression" because you did not send anything!

File: problems/5-memes/problem.md [hunk 2]
```diff
@@ -1,22 +1,53 @@
 # Memes for my eyes
 
-Any meme you upload is almost instantly available for retrieval. Other files also. Content is made available by root CID and by CAR shard CID. In this exercise you are going to fetch a text file from the public gateway by it's root CID and log it to the console.
+Ain't you cool? Now, let's shift gears and dive into the next challenge: retrieving content from a decentralized network.
 
-You can use the `w3s.link` gateway to retrieve the data but the magic of decentralized peer to peer networking means that you should be able to use any gateway.
+## The Challenge:
+Every meme (or any other file) you upload is instantly available for retrieval. In this exercise, your task is to fetch a text file from the public gateway using its root CID. The cool part? You can use any gateway, thanks to the decentralized nature of peer-to-peer networking. But for simplicity, we’ll use the w3s.link gateway.
```
Comment (alanshaw) [inline problems/5-memes/problem.md:6]:
```suggestion
Every meme (or any other file) you upload is instantly available for retrieval. In this exercise, your task is to fetch a text file from a public gateway using its root CID. The cool part? You can use any gateway, thanks to the decentralized nature of peer-to-peer networking. But for simplicity, we’ll use the `w3s.link` gateway.
```

File: problems/4-infinite-compression/problem.md [hunk 3]
```diff
@@ -1,22 +1,42 @@
...
+Let's move on.
+## What’s the Deal with Infinite Compression?
+Imagine compressing your data over and over until it’s reduced to its absolute minimum size. While true infinite compression is more of a thought experiment, in this challenge, you’ll get a taste of it by working with Content-Addressed Archives (CARs) and Directed Acyclic Graphs (DAGs).
+ 
+The DAG is packed into a CAR 🚗 - beep beep. CAR is kinda like a `tar` file but for DAGs. It's a useful container format for transferring content addressed data over HTTP.
+
+**Here’s how it works:** when you upload a file, it gets split into smaller pieces called CAR shards. Each of these shards gets a unique identifier called a CID (Content Identifier). When you’ve uploaded all the shards, they come together to form a DAG—a super-organized structure that ties everything up neatly with a root CID.
+
+## Your Task:
+1. Start by creating a new file, like ex5.mjs.
```
Comment (alanshaw) [inline problems/4-infinite-compression/problem.md:14]:
```suggestion
1. Start by creating a new file, like `ex5.mjs`.
```

File: problems/5-memes/problem.md [hunk 4]
```diff
@@ -1,22 +1,53 @@
...
+
+## What’s Happening Here?
+**CID Decoding:** The CID you receive is in a compact binary form. You use the multiformats library to decode it into a human-readable string that you can use in the URL.
+**Gateway Flexibility:** While we use w3s.link here, you can use any gateway that supports IPFS (InterPlanetary File System).
+**Real-time Retrieval:** Once you have the CID, fetching content from a decentralized network is as simple as sending a request to a URL.
+
+## Why This is Cool:
+This challenge shows you how easy it is to retrieve content from a decentralized network. It’s like pulling a meme out of thin air—instant access, no matter where the file is stored on the network.
+
+Now, create that ex6.mjs file, plug in your code, and see how seamlessly you can fetch content from the decentralized web. Happy coding! 🚀
```
Comment (alanshaw) [inline problems/5-memes/problem.md:43]:
```suggestion
Now, create that `ex6.mjs` file, plug in your code, and see how seamlessly you can fetch content from the decentralized web. Happy coding! 🚀
```

File: problems/5-memes/problem.md [hunk 5]
```diff
@@ -1,22 +1,53 @@
...
 
-To retrieve data from a gateway you should format the request like:
+2. Fetch the CID from `stdin`: The CID (Content Identifier) of the file you need to fetch will be provided via `process.stdin`. You’ll need to read this input first.
+```js
+import fs from 'node:fs'
+import { CID } from 'multiformats'
 
+const cid = CID.decode(fs.readFileSync(process.stdin.fd))
 ```
+3. Construct the Gateway URL: Format the URL to fetch the content using the w3s.link gateway. The URL format is:
```
Comment (alanshaw) [inline problems/5-memes/problem.md:18]:
```suggestion
3. Construct the Gateway URL: Format the URL to fetch the content using the `w3s.link` gateway. The URL format is:
```

File: problems/5-memes/problem.md [hunk 6]
```diff
@@ -1,22 +1,53 @@
...
+const res = await fetch(url)
+```
+5. Output the Content: Finally, write the fetched content directly to `process.stdout`.
+```js
+process.stdout.write(await res.text())
+```
+
+## What’s Happening Here?
+**CID Decoding:** The CID you receive is in a compact binary form. You use the multiformats library to decode it into a human-readable string that you can use in the URL.
+**Gateway Flexibility:** While we use w3s.link here, you can use any gateway that supports IPFS (InterPlanetary File System).
```
Comment (alanshaw) [inline problems/5-memes/problem.md:37]: They are typically called IPFS gateways - they'r enot a gateway for anything else.

```suggestion
**Gateway Flexibility:** We use w3s.link here, but you can use any IPFS gateway.
```

File: problems/6-dude-where/problem.md [hunk 7]
```diff
@@ -1,41 +1,59 @@
...
-
-One answer is to move the re-assembily from DAG to file into the client - the entity requesting the data. This removes the trust element as the client can now verify the hashes of the nodes in the graph.
-
-In this exercise you're going to fetch a CAR file from a "trustless" gateway (I know that sounds bad but actually it's good!) and re-assemble it into a file.
-
-Create a _new_ file for your solution e.g. `ex7.mjs`. You're going to need a couple of libraries, one for reading CARs and one for exporting files from an IPFS DAG:
-
-```sh
+## What You Need to Do:
+1. Yes, you guessed it - file number 7 - ex7.mjs, for this challenge.
```
Comment (alanshaw) [inline problems/6-dude-where/problem.md:9]:
```suggestion
1. Yes, you guessed it - file number 7 - `ex7.mjs`, for this challenge.
```

File: problems/6-dude-where/problem.md [hunk 8]
```diff
@@ -1,41 +1,59 @@
...
+2. Install Necessary Libraries: You’ll need two libraries to get this done: @ipld/car for reading CAR files and ipfs-unixfs-exporter for exporting files from an IPFS DAG.
```
Comment (alanshaw) [inline problems/6-dude-where/problem.md:10]:
```suggestion
2. Install Necessary Libraries: You’ll need two libraries to get this done: `@ipld/car` for reading CAR files and `ipfs-unixfs-exporter` for exporting files from an IPFS DAG.
```

File: problems/6-dude-where/problem.md [hunk 9]
```diff
@@ -1,41 +1,59 @@
...
+```js
+const cid = 'bafybeifdnnn35jnwel2dno7hu7zfew6gtxhkaynaszuj4i77j4m7wsv72a'
+const path = '/cat2.txt'
 
-You can fetch a CAR from the web3.storage trustless gateway like this:
-
+const res = await fetch(`https://dag.w3s.link/ipfs/${cid}?format=car`)
+const bytes = new Uint8Array(await res.arrayBuffer())
+```
+4. Read the CAR File: Use CarReader from @ipld/car to read the CAR file. This step is crucial because you need to adapt this reader to work with the exporter from ipfs-unixfs-exporter.
```
Comment (alanshaw) [inline problems/6-dude-where/problem.md:23]:
```suggestion
4. Read the CAR File: Use CarReader from `@ipld/car` to read the CAR file. This step is crucial because you need to adapt this reader to work with the exporter from `ipfs-unixfs-exporter`.
```

File: problems/7-commp/problem.md [hunk 10]
```diff
@@ -2,30 +2,59 @@
 
 Great job 👏 You are smashing it. Keep on truckin'.
 
-So, there's this crypto incentivized, decentralized file storage network called Filecoin. It's built on top of IPFS protocols and provides data storage for content addressed data. Maybe you heard of it?
+So, lets talk Filecoin...
 
-As you know, when you upload data using w3up you generate a DAG and a corresponding root CID. You also now know that w3up packs the DAG into 1 or more CAR files (shards). What you might _not_ know is that w3up also computes a **Piece CID** for each shard.
+## The Challenge
+The Challenge:
```
Comment (alanshaw) [inline problems/7-commp/problem.md:8]:
```suggestion

```

File: problems/7-commp/problem.md [hunk 11]
```diff
@@ -2,30 +2,59 @@
...
+Filecoin is a decentralized storage network built on IPFS protocols. When you upload data using w3up, it generates a DAG and its corresponding root CID. This DAG is packed into CAR files (shards), and each shard gets a Piece CID. This Piece CID is crucial because it’s how your data is addressed in deals with Filecoin Storage Providers (SPs).
 
-You can think of a piece CID as an alias for a shard. It is the primary means of addressing data that has been stored in deals made with Filecoin Storage Providers (SPs).
+The fun part? These deals are like massive shipping containers that need to be filled up. Your data goes through an aggregation pipeline, which ultimately leads to it being stored in a deal with a Filecoin SP. The process includes inclusion proofs and data aggregation proofs—basically, all the cryptographic magic that proves your data is securely stored and retrievable.
 
-When you upload data, the w3up client "offers" piece CIDs for filecoin deals. It invokes a `filecoin/offer` capability.
+## What You’re Going to Do:
+1. For the last time... ex8.mjs, you know the drill.
```
Comment (alanshaw) [inline problems/7-commp/problem.md:14]:
```suggestion
1. For the last time... `ex8.mjs`, you know the drill.
```

File: problems/7-commp/problem.md [hunk 12]
```diff
@@ -2,30 +2,59 @@
...
+2. Install the Necessary Libraries: You’ll need the @web3-storage/data-segment for parsing the Piece CID and @ipld/dag-json for encoding the output.
+```js
```
Comment (alanshaw) [inline problems/7-commp/problem.md:16]: It's not JS here!
```suggestion
```
```

**Review by alanshaw — CHANGES_REQUESTED** [review]

**Review by hakierka — COMMENTED** [review]
Commited suggested changes, changed the Infinite Compression copy - to be reviewed.

**Review by alanshaw — APPROVED** [review]
Just one small thing to consider but otherwise this LGTM.

File: problems/4-infinite-compression/problem.md [hunk 13]
```diff
@@ -1,22 +1,47 @@
 # Infinite avatar compression!
 
-YES BOSS! That's how you do it. Now your minions can do tasks on your behalf. They can even re-delegate to others! It's delegations all the way down.
+YAS BOSS! That's how you do it! 
 
-One thing to note about delegations is that they're not secrets, they can only be used by the target "audience" - in this case the DID of the workshop. So you can embed in scripts and/or send them in emails or whatever really. They're just a proof that you have been given the capability to do something.
+## What’s the Deal with Content Addressing?
+So, here’s the scoop: when you upload data to Storacha (previously web3.storage), it gets transformed into a **Directed Acyclic Graph (DAG)**. Each piece of your data becomes a node, and each node gets a unique identifier called a **Content Identifier (CID)**. This whole DAG is then packed into a Content-Addressed Archive (CAR)—think of it like a digital suitcase for your data, ready to travel the web.
```
Comment (alanshaw) [inline problems/4-infinite-compression/problem.md:6]: > Storacha (previously web3.storage)

I would leave as web3.storage and then do a separate PR to rename eveything.
